Doc/c-api/exceptions.rst

```diff
@@ -125,6 +125,44 @@ For convenience, some of these functions will always return a
 
    .. versionadded:: 3.5
 
+.. c:function:: void PyErr_ChainExceptions(PyObject *exc, PyObject *val, PyObject *tb)
+
+   Set the provided exception as the cause for the current exception. If there
+   is no exception currently set (:c:func:`PyErr_Occurred` returns ``0``) this
+   function behaves as like :c:func:`PyErr_Restore`, but if an exception is
+   already set, set the provided exception as the context (accessible from
+   Python through the :attr:`__context__` attribute of the exception) of the
+   one that is currently set.
+
+   The exception type should be a class. Do not pass an invalid exception type
+   or value. This function steals a reference to each of its arguments in the
```

Comment:
Could it *not* steal its arguments?
Functions that don't follow the usual calling convention are harder to reason about; it would be good to treat them as historical exceptions.

Reply:
It steals the arguments because of what it calls. It had the same calling convention as `Pyerr_Restore`.

Reply:
Also, stealing arguments is not an uncommon calling convention.